### ISSUE TITLE
Do not report Yoda condition problem where both values are de facto constants (__FILE__ == $0)

### ISSUE THREAD

**Issue by matkoniecz** [rubocop-hq/rubocop#7569]
Yoda conditions should not be reported where both values are treated as constants.
I think that $0 can be assumed to be constant, and modifying it reported as a separate cop.

--------

## Expected behavior

In a reasonable program `$0` is not going to be changed and both can be treated as read-only. So Yoda condition report should not be triggered for them.

## Actual behavior

```
C: Style/YodaCondition: Reverse the order of the operands __FILE__ == $0.
if __FILE__ == $0
```

## Steps to reproduce the problem

```
puts "test" if __FILE__ == $0
```

## RuboCop version

```
rubocop -V
`0.77.0 (using Parser 2.6.5.0, running on ruby 2.6.5 x86_64-linux-gnu)`
```

it was after

```
gem2.6 install --user --update rubocop
Fetching rubocop-0.77.0.gem
Successfully installed rubocop-0.77.0
1 gem installed
```
